yt/sample_data/api.py

```diff
@@ -143,7 +143,7 @@ def _get_test_data_dir_path():
     return Path.cwd()
 
 
-def lookup_on_disk_data(fn):
+def lookup_on_disk_data(fn) -> Path:
```

Comment:
```suggestion
from typing import Union
def lookup_on_disk_data(fn: Union[str, Path]) -> Path:
```

Reply:
I did not add a type here on purpose, become "filelike" is a little more complicated than this and I'd prefer adding a consistent `FileLike` type alias later at the level of the code base (see #3064)

Reply:
@cphyc , if you're contempt with my response here I think it's ready to go, otherwise let's discuss it further ?

Reply:
That's fine! Mergin'